internal/k8s/ingressstatus.go

```diff
@@ -0,0 +1,110 @@
+// Copyright © 2020 VMware
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package k8s
+
+import (
+	"github.com/sirupsen/logrus"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/api/networking/v1beta1"
+	clientset "k8s.io/client-go/kubernetes"
+)
+
+// StatusLoadbalancerUpdater observes informer OnAdd events and
+// updates the ingress.status.loadBalancer field on all Ingress
+// objects that match the ingress class (if used).
+type IngressStatusUpdater struct {
+	Client clientset.Interface
+	Logger logrus.FieldLogger
+	Status v1.LoadBalancerStatus
+}
+
```

Comment:
Nit: consider asserting that this implements `cache.ResourceEventHandler`. 
```
var _ cache.ResourceEventHandler = &IngressStatusUpdater{}
```

Reply:
I don't see what value this adds - if someone changes this struct to not implement that interface, then it will be a compile-time error.

Reply:
Fair enough. I started trying to get into the habit of doing this because it guarantees an error close to the definition and is robust against other code changes. We haven't been doing in in Contour, so not a big deal.